### PR TITLE
sys/log: Remove support for LOG_VERSION 2

### LINKED ISSUE #2120: Get rid of LOG_VERSION 2
More information here: http://mail-archives.apache.org/mod_mbox/mynewt-dev/201911.mbox/%3cCALy_Pv7S0897gwkaTtY0F_7S_aE5qUxUFa1bgdmkxcLFQPxUJA@mail.gmail.com%3e

### PR DESCRIPTION
**Pull request by vrahane**
- This fixes #2120. If you would like to know more, please take a look at the issue. 
- This also removes remnants of log_fcb_slot1.

### CODE REVIEW

File: sys/log/full/pkg.yml [hunk 1]
```diff
@@ -30,20 +30,18 @@ pkg.deps:
...
-pkg.deps.'LOG_CLI && LOG_VERSION > 2':
-    - "@apache-mynewt-core/sys/log/util/log_cbor_reader"
-
 pkg.deps.LOG_FCB:
     - "@apache-mynewt-core/hw/hal"
     - "@apache-mynewt-core/fs/fcb"
 pkg.deps.LOG_FCB2:
     - "@apache-mynewt-core/fs/fcb2"
 pkg.deps.LOG_CLI:
+    - "@apache-mynewt-core/sys/log/util/log_cbor_reader"
```
Comment (ccollins476ad) [inline sys/log/full/pkg.yml]: Was this addition intentional?

Comment (vrahane) [reply sys/log/full/pkg.yml]: yes, this is intentional. It was
```
pkg.deps.'LOG_CLI && LOG_VERSION > 2':
    - "@apache-mynewt-core/sys/log/util/log_cbor_reader"
```
which I removed above and combined it below.
